Test failure with macos-15-intel, enabled in os.yml

https://github.com/diofant/python-gmp/commit/d855ff987c835c419fd75644217e5bb67066875e

https://github.com/diofant/python-gmp/actions/runs/19202030080/job/54894380293

```
============================= test session starts ==============================
platform darwin -- Python 3.14.0, pytest-9.0.0, pluggy-1.6.0 -- /Library/Frameworks/Python.framework/Versions/3.14/bin/python

  The GNU GMP version: 6.3.0
  Bits per limb:       64
  sizeof(mp_limb_t):   8
  sizeof(mp_size_t):   8
  sizeof(mp_bitcnt_t): 8

cachedir: .pytest_cache
hypothesis profile 'default' -> deadline=timedelta(milliseconds=300000)
rootdir: /Users/runner/work/python-gmp/python-gmp
configfile: pyproject.toml
plugins: hypothesis-6.147.0
collecting ... collected 66 items / 2 skipped

Fatal Python error: Bus error

Current thread 0x00007ff85d883000 (most recent call first):
  File "/Users/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 40 in test_isqrt
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 999 in test
  File "/Users/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 36 in test_isqrt
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 1090 in run
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 817 in default_executor
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 1133 in execute_once
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 1193 in _execute_once_for_engine
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/internal/conjecture/engine.py", line 420 in __stoppable_test_function
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/internal/conjecture/engine.py", line 526 in test_function
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/internal/conjecture/engine.py", line 1261 in generate_new_examples
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/internal/conjecture/engine.py", line 1503 in _run
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/internal/conjecture/engine.py", line 933 in run
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 1381 in run_engine
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/hypothesis/core.py", line 2076 in wrapped_test
  File "/Users/runner/work/python-gmp/python-gmp/tests/test_functions.py", line 36 in test_isqrt
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/python.py", line 166 in pytest_pyfunc_call
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/python.py", line 1720 in runtest
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 179 in pytest_runtest_call
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 245 in <lambda>
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 353 in from_call
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 244 in call_and_report
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 137 in runtestprotocol
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/runner.py", line 118 in pytest_runtest_protocol
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/main.py", line 396 in pytest_runtestloop
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/main.py", line 372 in _main
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/main.py", line 318 in wrap_session
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/main.py", line 365 in pytest_cmdline_main
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/config/__init__.py", line 197 in main
  File "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/_pytest/config/__init__.py", line 221 in console_main
  File "/Users/runner/hostedtoolcache/Python/3.14.0/x64/bin/pytest", line 7 in <module>

Current thread's C stack trace (most recent call first):
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _Py_DumpStack+0x42 [0x1050fe0f2]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at faulthandler_dump_c_stack+0x45 [0x105113215]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at faulthandler_fatal_error+0x155 [0x105113135]
  Binary file "/usr/lib/system/libsystem_platform.dylib", at _sigtramp+0x1d [0x7ff81bec531d]
  Binary file '<unknown>' [0x10470d0e0]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/libzz.dylib", at zz_sqrtrem+0x194 [0x105a2f704]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/gmp.cpython-314-darwin.so", at gmp_isqrt+0xac [0x105a62ccc]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x212fa [0x104fc698a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyObject_Call_Prepend+0x14e [0x104e0f10e]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at slot_tp_call+0x36a [0x104ed7daa]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x271b2 [0x104fcc842]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyObject_Call_Prepend+0x14e [0x104e0f10e]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at slot_tp_call+0x36a [0x104ed7daa]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x219e0 [0x104fc7070]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyObject_Call_Prepend+0x14e [0x104e0f10e]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at slot_tp_call+0x36a [0x104ed7daa]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x271b2 [0x104fcc842]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyObject_Call_Prepend+0x14e [0x104e0f10e]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at slot_tp_call+0x36a [0x104ed7daa]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x271b2 [0x104fcc842]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyObject_Call_Prepend+0x14e [0x104e0f10e]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at slot_tp_call+0x36a [0x104ed7daa]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_EvalFrameDefault+0x271b2 [0x104fcc842]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at _PyEval_Vector+0x2aa [0x104fa2a5a]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at PyEval_EvalCode+0xa1 [0x104fa2721]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at run_mod+0x13b [0x1050dd57b]
  Binary file "/Library/Frameworks/Python.framework/Versions/3.14/Python", at pyrun_file+0x7a [0x1050dd33a]
  <truncated rest of calls>

Extension modules: gmp (total: 1)
/Users/runner/work/_temp/7cb06b91-d92f-4d70-b971-5d904deb2d6c.sh: line 1: 96867 Bus error: 10           pytest
tests/test_functions.py::test_isqrt 
Error: Process completed with exit code 138.
```

BTW, no failures in wheels.yml.

--- a/.github/workflows/os.yml
+++ b/.github/workflows/os.yml
@@ -6,7 +6,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-24.04, ubuntu-24.04-arm, macos-15]
+        os: [ubuntu-24.04, ubuntu-24.04-arm, macos-15-intel, macos-15]
         shell: [bash]
         include:
           - os: windows-2022
@@ -17,6 +17,7 @@ jobs:
     env:
       PYTEST_ADDOPTS: --verbose --hypothesis-profile=default
       CFLAGS: -Wall -Wpedantic -Werror -std=c17 -Wconversion
+      MACOSX_DEPLOYMENT_TARGET: 10.15  # for 3.14
     steps:
       - uses: actions/checkout@v5
         with:

